framework/src/interface/packet.rs

```diff
@@ -436,12 +436,16 @@ impl<T: EndOffset, M: Sized + Send> Packet<T, M> {
     pub fn insert_header_fn<T2: EndOffset<PreviousHeader = T>>(
         &mut self,
         header: &T2,
-        on_insert: &Fn(&mut T, isize),
-    ) -> Result<isize> {
+        on_insert: &Fn(&mut T),
+    ) -> Result<isize>
+    where
+        T: HeaderUpdates,
+    {
         unsafe {
             if let Ok(diff) = self.insert_header(header) {
                 let current_header: &mut T = &mut *self.header();
-                on_insert(current_header, diff);
+                current_header.update_payload(diff);
```

Comment:
ought to add this to the non-HOF versions too. Also, if we're doing this, probably don't need to return a `Result<isize>` from these mutation functions, can just essentially return a `Result<()>`, since it's no longer expected that the caller will need to use that value.